packages/flame/lib/src/components/mixins/has_hitboxes.dart

```diff
@@ -1,12 +1,38 @@
 import 'dart:collection';
-import 'dart:ui';
+import 'dart:math';
 
+import 'package:flutter/cupertino.dart';
+
+import '../../../components.dart';
 import '../../../extensions.dart';
-import '../../geometry/shape.dart';
-import '../position_component.dart';
+import '../../../geometry.dart';
 
 mixin HasHitboxes on PositionComponent {
   final List<HitboxShape> _hitboxes = <HitboxShape>[];
+  final Aabb2 _aabb = Aabb2();
+  bool _validAabb = false;
+  final Vector2 _maxHalfExtents = Vector2.zero();
+
+  @override
+  @mustCallSuper
+  void update(double dt) {
+    super.update(dt);
+    _validAabb = false;
```

Comment:
Looks like this invalidates the computed bounding rect on every game tick... 
However, if an object isn't moving then its bounding rect can be kept in cache.

Reply:
Indeed, we have no cheap way to check whether the component, or one of its ancestors, has gained a different size, scaled or moved.
If you know of a way that is cheaper than just re-creating the AABB I'm all ears, this will be even more important later when I turn it into a quad tree.

Reply:
I think for a start we could compute all bounding rects in world coordinates rather than screen coordinates. After all, there are plenty of objects that are stationary with respect to the world, but not so many that are stationary relative to the screen (other than HUD elements, but those don't usually collide :)

Reply:
All bounding rects are already in world coordinates.